code/game/machinery/dance_machine.dm

```diff
@@ -0,0 +1,479 @@
+	interact(user)
+
+/obj/machinery/disco/interact(mob/user)
+	if(!anchored)
+		to_chat(user,"<span class='warning'>This device must be anchored by a wrench!</span>")
+		return
+	if(!Adjacent(user) && !isAI(user))
+		return
+	user.set_machine(src)
+	var/list/dat = list()
+	dat +="<div class='statusDisplay' style='text-align:center'>"
+	dat += "<b><A href='?src=[UID()];action=toggle'>[!active ? "BREAK IT DOWN" : "SHUT IT DOWN"]<b></A><br>"
+	dat += "</div><br>"
+	dat += "<A href='?src=[UID()];action=select'> Select Track</A><br>"
+	dat += "Track Selected: [selection.song_name]<br>"
+	dat += "Track Length: [DisplayTimeText(selection.song_length)]<br><br>"
+	dat += "<br>DJ's Soundboard:<b><br>"
+	dat +="<div class='statusDisplay'><div style='text-align:center'>"
+	dat += "<A href='?src=[UID()];action=horn'>Air Horn</A>  "
+	dat += "<A href='?src=[UID()];action=alert'>Station Alert</A>  "
+	dat += "<A href='?src=[UID()];action=siren'>Warning Siren</A>  "
+	dat += "<A href='?src=[UID()];action=honk'>Honk</A><br>"
+	dat += "<A href='?src=[UID()];action=pump'>Shotgun Pump</A>"
+	dat += "<A href='?src=[UID()];action=pop'>Gunshot</A>"
+	dat += "<A href='?src=[UID()];action=saber'>Esword</A>"
+	dat += "<A href='?src=[UID()];action=harm'>Harm Alarm</A>"
+	var/datum/browser/popup = new(user, "vending", "Radiance Dance Machine - Mark IV", 400, 350)
+	popup.set_content(dat.Join())
+	popup.open()
+
+
+/obj/machinery/disco/Topic(href, href_list)
+	if(..())
+		return
+	add_fingerprint(usr)
+	switch(href_list["action"])
+		if("toggle")
+			if(qdeleted(src))
+				return
+			if(!active)
+				if(stop > world.time)
+					to_chat(usr, "<span class='warning'>Error: The device is still resetting from the last activation, it will be ready again in [DisplayTimeText(stop-world.time)].</span>")
+					playsound(src, 'sound/misc/compiler-failure.ogg', 50, 1)
+					return
+				active = TRUE
+				update_icon()
+				dance_setup()
+				processing_objects.Add(src)
+				lights_spin()
+				updateUsrDialog()
+			else if(active)
+				stop = 0
+				updateUsrDialog()
+		if("select")
+			if(active)
+				to_chat(usr, "<span class='warning'>Error: You cannot change the song until the current one is over.</span>")
+				return
+
+			var/list/available = list()
+			for(var/datum/track/S in songs)
+				available[S.song_name] = S
+			var/selected = input(usr, "Choose your song", "Track:") as null|anything in available
+			if(qdeleted(src) || !selected || !istype(available[selected], /datum/track))
+				return
+			selection = available[selected]
+			updateUsrDialog()
+		if("horn")
+			deejay('sound/items/airhorn2.ogg')
+		if("alert")
+			deejay('sound/misc/notice1.ogg')
+		if("siren")
+			deejay('sound/machines/engine_alert1.ogg')
+		if("honk")
+			deejay('sound/items/bikehorn.ogg')
+		if("pump")
+			deejay('sound/weapons/shotgunpump.ogg')
+		if("pop")
+			deejay('sound/weapons/gunshot3.ogg')
+		if("saber")
+			deejay('sound/weapons/saberon.ogg')
+		if("harm")
+			deejay('sound/ai/harmalarm.ogg')
```

Comment:
This causes compile failure on paradise-test:
code\game\machinery\dance_machine.dm:165:error: 'sound/ai/harmalarm.ogg': cannot find file

Reply:
This is what happens when you hastily revert content without a second thought in mind.